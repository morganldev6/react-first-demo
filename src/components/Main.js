import React from "react";
import Movie from "./Movie";
import uuid from "uuid/v4";

const Main = ({ movies }) => {
  console.log(movies);
  return (
    <div className="wrapper">
      {movies.map(movie => {
        return <Movie movie={movie} key={uuid()} />;
      })}
    </div>
  );
};

export default Main;
