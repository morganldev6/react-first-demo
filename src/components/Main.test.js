import React from "react";
import { render, cleanup } from "@testing-library/react";

import Main from "./Main";

afterEach(cleanup);

describe("Footer component", () => {
  const movies = [
    {
      acf: {
        release_year: "2015",
        rating: "4.3",
        description: "<p>Something about Star Wars Episode VII</p>"
      },
      title: { rendered: "Star Wars: Episode VII — The Force Awakens" }
    },
    {
      acf: {
        release_year: "1983",
        rating: "4.8",
        description: "<p>Something about Star Wars Episode VI</p>"
      },
      title: { rendered: "Star Wars: Episode VI — Return of the Jedi" }
    }
  ];
  test("Renders correctly", () => {
    const { container } = render(<Main movies={movies} />);
    expect(container).toMatchSnapshot();
  });
});
