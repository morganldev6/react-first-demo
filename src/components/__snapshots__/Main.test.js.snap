// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Renders correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="movieBlock"
    >
      <div
        class="contentBlock"
      >
        <h2>
          Title
          :
        </h2>
        <p>
          Star Wars: Episode VII — The Force Awakens
        </p>
      </div>
      <div
        class="contentBlock"
      >
        <h2>
          Release Year
          :
        </h2>
        <p>
          2015
        </p>
      </div>
      <div
        class="contentBlock"
      >
        <h2>
          Rating
          :
        </h2>
        <p>
          4.3
        </p>
      </div>
      <button
        class="button"
        type="button"
      >
        View/Hide Description
      </button>
    </div>
    <div
      class="movieBlock"
    >
      <div
        class="contentBlock"
      >
        <h2>
          Title
          :
        </h2>
        <p>
          Star Wars: Episode VI — Return of the Jedi
        </p>
      </div>
      <div
        class="contentBlock"
      >
        <h2>
          Release Year
          :
        </h2>
        <p>
          1983
        </p>
      </div>
      <div
        class="contentBlock"
      >
        <h2>
          Rating
          :
        </h2>
        <p>
          4.8
        </p>
      </div>
      <button
        class="button"
        type="button"
      >
        View/Hide Description
      </button>
    </div>
  </div>
</div>
`;
